platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/builders/TaskSchedulerConfiguration.java

```diff
@@ -0,0 +1,141 @@
+     *     </li>
+     * </ul>
+     * Example: "SEQUENTIAL CAPACITY(500) !FLUSHABLE UNHANDLED_TASK_METRIC"
+     * <p>
+     * Note that default values are not specified within this class. Default values are the responsibility of the
+     * {@link TaskSchedulerBuilder} class.
+     *
+     * @param string the string to parse
+     * @return the parsed configuration
+     */
+    @NonNull
+    public static TaskSchedulerConfiguration parse(@NonNull final String string) {
+        TaskSchedulerType type = null;
+        Long unhandledTaskCapacity = null;
+        Boolean unhandledTaskMetricEnabled = null;
+        Boolean busyFractionMetricEnabled = null;
+        Boolean flushingEnabled = null;
+        Boolean squelchingEnabled = null;
+
+        try {
+            final String[] parts = string.split(" ");
+            for (final String part : parts) {
+                String strippedPart = part.strip();
+                if (strippedPart.isEmpty()) {
+                    continue;
+                }
+
+                try {
+                    // if the part can be parsed as a TaskSchedulerType, then it is the type
+                    type = TaskSchedulerType.valueOf(strippedPart);
+                    continue;
+                } catch (final IllegalArgumentException e) {
+                    // ignore
+                }
+
+                if (strippedPart.startsWith(TaskSchedulerConfigOption.CAPACITY)) {
+                    try {
+                        // parse a string in the form "CAPACITY(1234)"
+                        final int openParenIndex = strippedPart.indexOf('(');
+                        final int closeParenIndex = strippedPart.indexOf(')');
+                        if (openParenIndex == -1 || closeParenIndex == -1) {
+                            throw new IllegalArgumentException("Invalid capacity \"" + strippedPart + "\"");
+                        }
+                        final String capacityString = strippedPart.substring(openParenIndex + 1, closeParenIndex);
+                        unhandledTaskCapacity = Long.parseLong(capacityString);
+                        continue;
+                    } catch (final NumberFormatException e) {
+                        throw new IllegalArgumentException("Invalid capacity \"" + strippedPart + "\"", e);
+                    }
+                }
+
+                try {
+                    // All other values must be TaskSchedulerConfigOption values.
+                    boolean value = true;
+                    if (strippedPart.startsWith("!")) {
+                        value = false;
+                        strippedPart = strippedPart.substring(1);
+                    }
+
+                    final TaskSchedulerConfigOption option = TaskSchedulerConfigOption.valueOf(strippedPart);
+
+                    switch (option) {
+                        case UNHANDLED_TASK_METRIC -> unhandledTaskMetricEnabled = value;
```

Comment:
consider throwing if any value already defined is parsed again

Reply:
check added